subprojects/tooling-api-builders/src/main/java/org/gradle/tooling/internal/provider/runner/TestExecutionBuildConfigurationAction.java

```diff
@@ -57,6 +58,14 @@ private void configureTestTasks(Set<Test> allTestTasksToRun) {
             task.setIgnoreFailures(true);
             task.getFilter().setFailOnNoMatchingTests(false);
             task.getOutputs().upToDateWhen(Specs.SATISFIES_NONE);
+            InternalDebugOptionsRequest debugOptions = testExecutionRequest.getDebugOptions();
+            if (debugOptions.isDebugMode()) {
+                // TODO (donat) remove existing debug options from JVM arguments
+                List<String> jvmArgs = new ArrayList<>(task.getJvmArgs().size() + 1);
+                jvmArgs.addAll(task.getJvmArgs());
+                jvmArgs.add("-agentlib:jdwp=transport=dt_socket,server=y,suspend=" + (debugOptions.isSuspend() ? "y" : "n") + ",address=" + debugOptions.getPort());
```

Comment:
We should probably have a method for that on the test task instead of having the logic here.

Reply:
👍

Reply:
I've added a new `setDebugOptions` to `JvmOptions`.